beacon-chain/rpc/statefetcher/fetcher.go

```diff
@@ -16,6 +16,26 @@ import (
 	"github.com/prysmaticlabs/prysm/shared/bytesutil"
 )
 
+// ErrInvalidStateId represents an error scenario where a state ID is invalid.
+var ErrInvalidStateId = errors.New("invalid state ID")
+
+// StateNotFoundError represents an error scenario where a state could not be found.
+type StateNotFoundError struct {
```

Comment:
I created a new type to be able to compare errors. This error has a formatted string, so using `var errFoo = errors.New("foo")` would not work.